packages/m2m-gateway/test/vitest.api.setup.ts

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { vi } from "vitest";
+import { Request, Response, NextFunction } from "express";
+
+vi.mock("pagopa-interop-application-audit", async () => ({
+  applicationAuditBeginMiddleware: vi.fn(
+    async () => (_req: Request, _res: Response, next: NextFunction) => next()
+  ),
+  applicationAuditEndMiddleware: vi.fn(
+    async () => (_req: Request, _res: Response, next: NextFunction) => next()
+  ),
+}));
+
+vi.mock("pagopa-interop-commons", async () => {
+  const actual = await vi.importActual<typeof import("pagopa-interop-commons")>(
+    "pagopa-interop-commons"
+  );
+  return {
+    ...actual,
+    authenticationMiddleware: vi.fn(
+      () =>
+        async (
```

Comment:
```suggestion
  return {
    ...actual,
    initRedisRateLimiter: vi.fn().mockResolvedValue({
      rateLimiterMiddleware:
        () => (_req: Request, _res: Response, next: NextFunction) =>
          next(),
    }),
    authenticationMiddleware: vi.fn(
      () =>
        async (
```

Trying to run the tests locally I also had a problem where they were hanging forever. This has fixed it for me, not sure if it will also fix the CI.